### PR TITLE
[PRD-3023]-Named color code for elements is NOT the same for chart.

### PR DESCRIPTION
**Pull request by denisprotopopov**
- Added a feature-switch to the reporting configuration

### CODE REVIEW

**Comment by denisprotopopov** [comment]
@mbatchelor @tmorgner please review the changes

**Comment by denisprotopopov** [comment]
@mbatchelor @tmorgner please review the changes

File: engine/legacy-charts/src/main/java/org/pentaho/plugin/jfreereport/reportcharts/ColorHelper.java [hunk 1]
```diff
@@ -80,7 +81,13 @@ private ColorHelper() {
     colors.put( "gold", Color.decode( "#ffd700" ) ); //$NON-NLS-1$ //$NON-NLS-2$
     colors.put( "goldenrod", Color.decode( "#daa520" ) ); //$NON-NLS-1$ //$NON-NLS-2$
     colors.put( "gray", Color.decode( "#808080" ) ); //$NON-NLS-1$ //$NON-NLS-2$
-    colors.put( "green", Color.decode( "#008000" ) ); //$NON-NLS-1$ //$NON-NLS-2$
+    if ( !ClassicEngineBoot.getInstance().getExtendedConfig().
+      getBoolProperty( "org.pentaho.reporting.engine.classic.extensions.legacy.charts.LegacyColorMode" ) ) {
+      colors.put( "green", Color.decode( "#00ff00" ) ); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (tmorgner) [inline engine/legacy-charts/src/main/java/org/pentaho/plugin/jfreereport/reportcharts/ColorHelper.java]: So there is no "lime" color defined when legacy mode is off? What color will reports see when they previously defined a "lime" bar chart?

Comment (denisprotopopov) [inline engine/legacy-charts/src/main/java/org/pentaho/plugin/jfreereport/reportcharts/ColorHelper.java]: it will be red, as unknown color

Comment (tmorgner) [inline engine/legacy-charts/src/main/java/org/pentaho/plugin/jfreereport/reportcharts/ColorHelper.java]: why should it be red? The old reports with that color were't red, were they? Please fix that.

Comment (denisprotopopov) [inline engine/legacy-charts/src/main/java/org/pentaho/plugin/jfreereport/reportcharts/ColorHelper.java]: fixed

**Comment by denisprotopopov** [comment]
@mbatchelor @tmorgner please review the changes

**Pull request merged**